xbmc/windowing/win10/WinSystemWin10.cpp

```diff
@@ -333,8 +336,10 @@ bool CWinSystemWin10::ChangeResolution(const RESOLUTION_INFO& res, bool forceCha
         HdmiDisplayMode selected = nullptr;
         for (const auto& mode : hdmiModes)
         {
-          if (res.iScreenWidth == mode.ResolutionWidthInRawPixels() && res.iScreenHeight == mode.ResolutionHeightInRawPixels()
-            && fabs(res.fRefreshRate - mode.RefreshRate()) <= 0.00001)
+          if ((!needHDR || (needHDR && mode.ColorSpace() == HdmiDisplayColorSpace::BT2020)) &&
+              res.iScreenWidth == mode.ResolutionWidthInRawPixels() &&
+              res.iScreenHeight == mode.ResolutionHeightInRawPixels() &&
+              fabs(res.fRefreshRate - mode.RefreshRate()) <= 0.00001)
           {
             selected = mode;
```

Comment:
No need to test the mode SDR/HDR attribute to match needHDR, like the HDR toggle?

Reply:
Initially the code was:

```c++
const auto newColoSp = (needHDR) ? HdmiDisplayColorSpace::BT2020 : HdmiDisplayColorSpace::BT709;

if (mode.ColorSpace() == newColorSp && ...)
```

But this broken SDR 23,976 Hz refresh rate (introduces a regression even for users that not use HDR). Probably is because some video modes not has any color space defined `::None`. Then only is compared color space for HDR (BT.2020) but not for SDR. This is same as before.... 

`(!needHDR || (needHDR && mode.ColorSpace() == HdmiDisplayColorSpace::BT2020))`

This condition is always true when is not needHDR. In other words is not possible cause regressions because code used to switch SDR modes not changes.

Reply:
See my other comment about checking other attributes of mode instead of ColorSpace(), maybe it helps.

With needHDR = false, the test allows modes with any colorspace and good results seem depend SDR modes being enumerated first, which is not very robust, Kodi doesn't control the enumeration.

If no attribute can be used then please add a comment that SDR modes are assumed to be enumerated first, otherwise the wrong mode may be selected. The toggle code has the same weakness.

Reply:
These are all modes supported in Xbox Series S:

```
2023-11-13 16:55:28.931 T:788      info <general>: 1920x1080, 23.98 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.933 T:788      info <general>: 1280x720,  50.00 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.935 T:788      info <general>: 3840x2160, 59.94 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.937 T:788      info <general>: 1920x1080, 50.00 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.938 T:788      info <general>: 3840x2160, 23.98 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.940 T:788      info <general>: 3840x2160, 50.00 Hz, colorSpace: 0, Smpte2084: no, 2086Metadata: no, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.942 T:788      info <general>: 3840x2160, 59.94 Hz, colorSpace: 3, Smpte2084: no, 2086Metadata: yes, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.943 T:788      info <general>: 3840x2160, 59.94 Hz, colorSpace: 2, Smpte2084: yes, 2086Metadata: yes, DolbyVision: yes, Stereo: no
2023-11-13 16:55:28.944 T:788      info <general>: 3840x2160, 23.98 Hz, colorSpace: 2, Smpte2084: yes, 2086Metadata: yes, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.946 T:788      info <general>: 3840x2160, 23.98 Hz, colorSpace: 3, Smpte2084: no, 2086Metadata: yes, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.947 T:788      info <general>: 3840x2160, 50.00 Hz, colorSpace: 3, Smpte2084: no, 2086Metadata: yes, DolbyVision: no, Stereo: no
2023-11-13 16:55:28.949 T:788      info <general>: 3840x2160, 50.00 Hz, colorSpace: 2, Smpte2084: yes, 2086Metadata: yes, DolbyVision: yes, Stereo: no
```

2 = BT.2020
3 = BT.709
0 = RgbLimited

Some resolutions only exits in RGBLimited then not all "SDR modes" uses BT.709 
All HDR modes uses BT.2020 and is equivalent to check for Smpte2084 supported

Then I replaced this in HDR toggle code as seems safe.

BUT same replacement in ChangeResolution() cause side effects e.g. 59.94 Hz when HDR is already on and start playback of 23.974 fps movie.

Currently I do not understand the cause and unfortunately do not have time to investigate and doing multiple required tests. To change this I would have to delay it for another week and I think it is now more important to ensure that this is included in Beta2 and RC1 to have feedback and have time to fix possible critical errors before final version.

Reply:
OK so the RGB limited modes are srgb instead of BT709. The gamma curve difference can be compensated.

Fine to push as is with a comment explaining that block is still a work in progress.